mlflow/gateway/providers/anthropic.py

```diff
@@ -17,6 +17,7 @@ class AnthropicAdapter(ProviderAdapter):
     @classmethod
     def chat_to_model(cls, payload, config):
         key_mapping = {"stop": "stop_sequences"}
+        payload["model"] = config.model.name
```

Comment:
note: the "model" key was set in the async hander before (see diff at L261-). Moving it inside the adapter as we want request transformation logic to complete within the adapter.